hicommonwealth/commonwealth
# Simplify Role handling, eliminating RolesController.

## Link to Issue
Closes: #8285

## Description of Changes
The `app.roles` controller maintained a `roles` list, which functioned as a unclear abstraction: it contained "role" objects for the current user only. Given that a role object is a simple permission string on top of an account object (as membership is determined by address+community_id on the account), but is functionally an entirely separate object, the goal of this PR was to eliminate all use of Roles on the client. 

Removing Roles involved enforcing two main contracts:
1. Communities must own their own list of mods and admins, as a list of addresses.
2. The UserStore must own the list of accounts owned by the currently logged in user, i.e. the canonical list of address x community pairings.

Given these two contracts, it should be straightforward to check a role at any time, reflected in the updates to the Permissions util.

The remaining work revolved around updating admin and moderator management / role assignment within a community. There are still outstanding product questions in this area, as we display profile names but ultimately upgrade individual addresses.

Finally, a change to remove "includeRoles" was made to searchProfiles, as it was no longer needed. Legacy Role objects and types were stripped from various places across the codebase alongside this change.

## Test Plan
- Create community
- Update community (http://localhost:8080/:communityId/manage/profile)
- Check that the admins and members management page works, you should be able to remove and add members and they should persist on page refresh (http://localhost:8080/:communityId/manage/moderators)
- Check that the you can create/update/delete groups (http://localhost:8080/:communityIdmembers/groups/create)
